tests/integration-tests/network_template_builder.py

```diff
@@ -158,6 +209,109 @@ def __build_vpc(self):
+                service_name=prefix + f"com.amazonaws.{region}.cloudformation",
+                type=VPCEndpointConfig.EndpointType.INTERFACE,
+                enable_private_dns=True,
+            ),
+            VPCEndpointConfig(
+                name="EC2Endpoint",
+                service_name=prefix + f"com.amazonaws.{region}.ec2",
+                type=VPCEndpointConfig.EndpointType.INTERFACE,
+                enable_private_dns=True,
+            ),
+            VPCEndpointConfig(
+                name="SecretsManager",
+                service_name=f"com.amazonaws.{region}.secretsmanager",
+                type=VPCEndpointConfig.EndpointType.INTERFACE,
+                enable_private_dns=True,
+            ),
+            VPCEndpointConfig(
+                name="S3Endpoint",
+                service_name=f"com.amazonaws.{region}.s3",
+                type=VPCEndpointConfig.EndpointType.GATEWAY,
+                enable_private_dns=False,
+            ),
+            VPCEndpointConfig(
+                name="DynamoEndpoint",
+                service_name=f"com.amazonaws.{region}.dynamodb",
+                type=VPCEndpointConfig.EndpointType.GATEWAY,
+                enable_private_dns=False,
+            ),
+        ]
+
+        for vpc_endpoint in vpc_endpoints:
+            vpc_endpoint_kwargs = {
+                "ServiceName": vpc_endpoint.service_name,
+                "PrivateDnsEnabled": vpc_endpoint.enable_private_dns,
+                "VpcEndpointType": str(vpc_endpoint.type),
+                "VpcId": Ref(self.__vpc),
+            }
+            if vpc_endpoint.type == VPCEndpointConfig.EndpointType.INTERFACE:
+                vpc_endpoint_kwargs["SubnetIds"] = [subnet_id]
+            if vpc_endpoint.type == VPCEndpointConfig.EndpointType.GATEWAY:
+                vpc_endpoint_kwargs["RouteTableIds"] = route_table_ids
+
+            self.__template.add_resource(
+                VPCEndpoint(
+                    vpc_endpoint.name,
+                    **vpc_endpoint_kwargs,
+                )
+            )
+
+    def __build_bastion_instance(self, bastion_subnet_id):
+        bastion_sg = ec2.SecurityGroup(
+            "NetworkingTestBastionSG",
+            GroupDescription="SecurityGroup for Bastion",
+            SecurityGroupIngress=[
+                ec2.SecurityGroupRule(
+                    IpProtocol="tcp",
+                    FromPort="22",
+                    ToPort="22",
+                    CidrIp="0.0.0.0/0",
+                ),
+            ],
+            VpcId=Ref(self.__vpc),
+        )
+        launch_template = ec2.LaunchTemplate.from_dict(
+            "LaunchTemplateIMDSv2",
+            {
+                "LaunchTemplateData": {"MetadataOptions": {"HttpTokens": "required", "HttpEndpoint": "enabled"}},
+            },
+        )
+        self.__template.add_resource(launch_template)
+        instance = ec2.Instance(
+            "NetworkingBastionInstance",
+            InstanceType="c5.xlarge",
```

Comment:
Do we want to hardcode here the instance type ?

Reply:
I'm going to create a constant